migrator/migrations/m_108_to_m_109_replace_deprecated_resources/migration.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/tecbot/gorocksdb"
+)
+
+const (
+	batchSize = 500
+
+	// Replacement resources
+	Access              = "Access"
+	Administration      = "Administration"
+	Compliance          = "Compliance"
+	DeploymentExtension = "DeploymentExtension"
+	Image               = "Image"
+	Integration         = "Integration"
+	// Replaced resources
+	AllComments           = "AllComments"
+	APIToken              = "APIToken"
+	AuthProvider          = "AuthProvider"
+	BackupPlugins         = "BackupPlugins"
+	ComplianceRuns        = "ComplianceRuns"
+	ComplianceRunSchedule = "ComplianceRunSchedule"
+	Config                = "Config"
+	DebugLogs             = "DebugLogs"
+	Group                 = "Group"
+	ImageComponent        = "ImageComponent"
+	ImageIntegration      = "ImageIntegration"
+	Indicator             = "Indicator"
+	Licenses              = "Licenses"
+	NetworkBaseline       = "NetworkBaseline"
+	NetworkGraphConfig    = "NetworkGraphConfig"
+	Notifier              = "Notifier"
+	ProbeUpload           = "ProbeUpload"
+	ProcessWhitelist      = "ProcessWhitelist"
+	Risk                  = "Risk"
+	Role                  = "Role"
+	ScannerBundle         = "ScannerBundle"
+	ScannerDefinitions    = "ScannerDefinitions"
+	SensorUpgradeConfig   = "SensorUpgradeConfig"
+	ServiceIdentity       = "ServiceIdentity"
+	SignatureIntegration  = "SignatureIntegration"
+	User                  = "User"
+)
+
+var (
+	migration = types.Migration{
+		StartingSeqNum: 108,
+		VersionAfter:   storage.Version{SeqNum: 109},
+		Run: func(databases *types.Databases) error {
+			return migatePermissionSets(databases.RocksDB)
+		},
+	}
+
+	prefix = []byte("permission_sets")
+
+	replacements = map[string]string{
+		AllComments:           Administration,
+		APIToken:              Integration,
+		AuthProvider:          Access,
+		BackupPlugins:         Integration,
+		ComplianceRuns:        Compliance,
+		ComplianceRunSchedule: Administration,
+		Config:                Administration,
+		DebugLogs:             Administration,
+		Group:                 Access,
+		ImageComponent:        Image,
+		ImageIntegration:      Integration,
+		Indicator:             DeploymentExtension,
+		Licenses:              Access,
+		NetworkBaseline:       DeploymentExtension,
+		NetworkGraphConfig:    Administration,
+		Notifier:              Integration,
+		ProbeUpload:           Administration,
+		ProcessWhitelist:      DeploymentExtension,
+		Risk:                  DeploymentExtension,
+		Role:                  Access,
+		ScannerBundle:         Administration,
+		ScannerDefinitions:    Administration,
+		SensorUpgradeConfig:   Administration,
+		ServiceIdentity:       Administration,
+		SignatureIntegration:  Integration,
+		User:                  Access,
+	}
+
+	readOpts  = gorocksdb.NewDefaultReadOptions()
+	writeOpts = gorocksdb.NewDefaultWriteOptions()
+)
+
+func propagatePermission(resource string, accessLevel storage.Access, permissions map[string]storage.Access) {
```

Comment:
Instead of modifying `permissions` (side effect), let's return updated `storage.Access`.